client/a8c-for-agencies/data/purchases/use-fetch-licenses.ts

```diff
@@ -0,0 +1,63 @@
+import config from '@automattic/calypso-config';
+import { useQuery } from '@tanstack/react-query';
+import { LICENSES_PER_PAGE } from 'calypso/a8c-for-agencies/sections/purchases/lib/constants';
+import {
+	LicenseFilter,
+	LicenseSortDirection,
+	LicenseSortField,
+} from 'calypso/jetpack-cloud/sections/partner-portal/types';
+
+export default function useFetchLicenses(
+	filter: LicenseFilter,
+	search: string,
+	sortField: LicenseSortField,
+	sortDirection: LicenseSortDirection,
+	page: number
+) {
+	const showDummyData = config.isEnabled( 'a4a/mock-api-data' );
+
+	const getLicenses = () => {
+		if ( showDummyData ) {
+			return {
+				items: [
```

Comment:
Can we also mock search filter here so we can test the search function?

Reply:
I didn't want to waste a lot of time on this, so I wanted to keep it simple. I think we can test it when we implement the actual API. What do you think?

Reply:
I think it is worth doing a simple product filtering here. There are lots of UI elements involved in the search functionality (e.g., no hit message, pagination, etc.).

Reply:
It would benefit us to be able to test it early without the API.

Reply:
Sounds good. I have added a https://github.com/Automattic/jetpack-genesis/issues/256 to address this. I feel like this can take some time, so we can address in a different PR. WDYT?